hxinsight-extension/src/test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionAnswersRelationTest.java

```diff
@@ -0,0 +1,61 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.hxi_extension.rest.api;
+
+import static org.mockito.BDDMockito.then;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.alfresco.hxi_connector.hxi_extension.service.HxInsightClient;
+
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert")
+class QuestionAnswersRelationTest
+{
+
+    @Mock
+    private HxInsightClient mockHxInsightClient;
+
+    @InjectMocks
+    private QuestionAnswersRelation objectUnderTest;
+
+    @Test
+    void shouldCallHxInsightClientGetAnswer()
+    {
+        // given
+        String questionId = "questionId";
+
+        // when
+        objectUnderTest.readAll(questionId, null);
+
+        // then
+        then(mockHxInsightClient).should().getAnswer(questionId);
```

Comment:
maybe you should also test if answer is passed to the caller?

Reply:
But it isn't yet. I'll do that when I work on the next task in the story. Will also add some more tests (including integration test I hope).